spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -79,6 +85,13 @@ public void onSuccess(String result) {
 	}
 
 	@Override
+	public Subscriber poll(String subscription, MessageReceiver messageHandler) {
+		Subscriber subscriber = this.subscriberFactory.getSubscriber(subscription, messageHandler);
+		subscriber.startAsync();
+		return subscriber;
```

Comment:
I somehow don't see a reason to have this functionality here. It just doesn't bring too match value and just fully delegates to the `subscriberFactory`. 
What is the point to overhead with this, please?

Reply:
In my mind, the PubSubOperation should be the one stop surface for commonly used Pub/Sub interactions, which is to publish, pull, subscribe. SubscriberFactory is a bit of a utility that helps to do it,  the basic functionality should be in the PubSubOperation rather than having to deal w/ SubscriberFactory (or other things).

Reply:
I tend to agree with @saturnism .
Using `PubSubOperations`/`PubSubTemplate` as a standalone abstraction, it isn't possible to pull messages currently.
Users have to directly use `SubscriberFactory`, which makes `PubSubOperations` incomplete as a Pub/Sub surface. I consider `SubscriberFactory` to be an implementation detail.

Reply:
Yeah. Agreed. Much easier to inject only one abstraction. Thank you and never mind!